Source/WTF/wtf/MemoryPressureHandler.cpp

```diff
@@ -268,6 +270,23 @@ void MemoryPressureHandler::endSimulatedMemoryPressure()
     memoryPressureStatusChanged();
 }
 
+void MemoryPressureHandler::beginSimulatedMemoryWarning()
+{
+    if (m_isSimulatingMemoryWarning)
```

Comment:
It may be worth adding ASSERTS to track why we are calling `beginSimulatedMemoryWarning` after already enabling it. Same for `endSimulatedMemoryWarning`.

Reply:
I disagree! That would mean you could crash your process by sending the notification twice via `notifyutil`. Highly undesirable behavior.

Reply:
To answer the question you were implying (what is causing this to be called when we're already simulating) - if you post back-to-back notifications like this:

```
$ notifyutil -p org.WebKit.memoryWarning.begin
$ notifyutil -p org.WebKit.memoryWarning.begin
```
 (same with `.end` and with the `lowMemory` memory pressure sim variant)
 
 If we assert then you could easily crash your process by accidentally doing the above.

Reply:
Ah, I didn't realize that the begin would stick around for a while and it could trigger it that easily.